pyxem/tests/signals/test_electron_diffraction2d.py

```diff
@@ -134,27 +134,23 @@ def test_set_scan_calibration(self, diffraction_pattern):
         assert isinstance(diffraction_pattern, ElectronDiffraction2D)
 
     @pytest.mark.parametrize(
```

Comment:
This could be simplified a little bit with the change.

Something like:
```python
    @pytest.mark.parametrize(
        "calibration, center, pixels",
        [
            (0.5, (4, 4), True),
            (0.5, (2, 2), False),
            (0.5, None, False),
        ],
    )
    def test_set_diffraction_calibration(
        self, diffraction_pattern, calibration, center, pixels
    ):
        diffraction_pattern.set_diffraction_calibration(
            calibration, center=center, pixels=pixels
        )
        dx, dy = diffraction_pattern.axes_manager.signal_axes
        assert dx.scale == calibration and dy.scale == calibration
        assert dx.offset == -2 and dy.offset == -2
```

Reply:
I think the test will need to be a little more complex than this. For instance, `dx.offset` will depend on the `center` and `pixels` values. In addition, I think it is a good idea to also check on the data values and make sure that the specified center matches what is retrieved on the data side when accessed through `dp.isig[...]`